client-app/src/desktop/tabs/other/routing/SimpleRoutingPanel.tsx

```diff
@@ -0,0 +1,90 @@
+import {grid, GridModel} from '@xh/hoist/cmp/grid';
+import {StoreRecordId} from '@xh/hoist/data';
+import {Icon} from '@xh/hoist/icon';
+import {panel} from '@xh/hoist/desktop/cmp/panel';
+import {wrapper} from '../../../common';
+import React from 'react';
+
+export const simpleRoutingPanel = hoistCmp.factory({
+    displayName: 'SimpleRoutingPanel',
+    model: creates(() => new SimpleRoutingPanelModel()),
+
+    render({model}) {
+        return wrapper({
+            description: [
+                <p>
+                    Hoist provides functionality for route parameters to interact with UI
+                    components. Below is a simple grid whose selected record can be maintained by a
+                    route parameter.{' '}
+                </p>,
+                <p>
+                    E.g. URLs like <code>https://toolbox.xh.io/app/other/simpleRouting/123</code>,
+                    where <code>123</code> is a record ID and that record is (auto) selected in the
+                    grid. Additionally, the route parameter will be updated when the user selects a
+                    different record in the grid.
+                </p>
+            ],
+            item: panel({
+                title: 'Simple Routing',
+                icon: Icon.gridPanel(),
+                className: 'tb-simple-routing-panel',
+                item: grid({model: model.gridModel}),
+                height: 600,
+                width: 600
+            })
+        });
+    }
+});
+
+class SimpleRoutingPanelModel extends HoistModel {
+    gridModel = new GridModel({
+        columns: [
+            {field: 'id', flex: 0},
+            {field: 'company', flex: 1}
+        ]
+    });
+
+    constructor() {
+        super();
+        this.addReaction({
+            track: () => XH.routerState.params,
+            run: () => this.updateGridSelectionOnRouteChange(),
+            fireImmediately: true
+        });
+        this.addReaction({
+            track: () => this.gridModel.selectedId,
+            run: () =>
+                this.updateRouteOnGridSelectionChange(
+                    XH.routerState.name,
+                    this.gridModel.selectedId
+                ),
+            fireImmediately: true
+        });
+    }
+
+    async updateGridSelectionOnRouteChange() {
+        if (
+            !XH.routerState.params.recordId ||
+            XH.routerState.params.recordId === this.gridModel.selectedId
+        )
+            return;
+        await this.gridModel.selectAsync(Number(XH.routerState.params.recordId));
+    }
+
+    updateRouteOnGridSelectionChange(name: string, selectedId: StoreRecordId) {
+        if (
+            !name.startsWith('default.other.simpleRouting') ||
+            !selectedId ||
+            XH.routerState.params.recordId === selectedId
+        )
+            return;
+        XH.navigate('default.other.simpleRouting.recordId', {recordId: selectedId});
```

Comment:
I initially decided to not `replace` because that would allow a user to hit the `back` button on their browser and view a prior selection. @amcclain thoughts?

Reply:
After talking to @amcclain we decided to have `{replace:true}` for this example and a more "advanced" implementation for the advanced routing example 👍